rfcs/0000-single-cluster-dnspolicy.md

```diff
@@ -0,0 +1,272 @@
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## API Updates
+
+DNSPolicy:
+
+- new providerRef field `spec.providerRef`
+- - new strategy field `spec.strategy` 
+
+DNSRecord:
+ 
+- `spec.managedZone` replaced with `spec.providerRef`
+- - new zoneID field `spec.zoneID`
+
+ManagedZone:
+
+- `spec.dnsProviderSecretRef` replaced with `spec.providerRef`
+
+### DNSPolicy.spec.providerRef
+
+The `providerRef` field is mandatory and contains a reference to a resource that shows how DNSRecords will be reconciled.
+    - `spec.providerRef.name` - name of the provider resource
+    - `spec.providerRef.kind` - kind of resource, can be anything but only `Secret` or `Managedzone` will be reconciled by MGC.
+
+A kind of type `Secret` will be managed by MGC, and a secret in the dns policies namespace with the given name must exist. The expected contents of the secrets data is comparable to the `dnsProviderSecretRef` used by ManageZones.
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: mgc-aws-credentials
+type: kuadrant.io/aws
+data:
+  AWS_ACCESS_KEY_ID: "foo"
+  AWS_SECRET_ACCESS_KEY: "bar"
+  CONFIG:
+    zoneIDFilter:
+      - Z04114632NOABXYWH93QUl
+```
+
+The `CONFIG` section of the secrets data will be added to allow provider specific configuration to be stored alongside the providers credentials and can be used during the instantiation of the provider client, and during any provider operations.
+The above for example would use the `zoneIDFilter` value to limit what hosted zones this provider is allowed to update.
+
+A kind of type `ManagedZone` will be managed by MGC, and a ManagedZone in the dns policies namespace with the given name must exist. 
+
+A kind of any other type e.g. `ExternalDNS` informs MGC that it should not reconcile any resources referencing it. All fields of `providerRef` will be ignored by MGC and can be set to anything, however since the `providerRef` will still be copied over to any DNSRecord resources created by the policy controller the values may still be given meaning to that external DNSRecord reconciler.
+
+### DNSPolicy.spec.strategy[simple|loadalanced]
+
+The `strategy` field is mandatory and dictates what kind of dns record structure the policy will create. Two strategy options are allowed `simple` or `loadbalanced`.
+
+A reconciliation of DNSPolicy processes the target gateway and creates a DNSRecord per listener that is supported by the currently configured provider(hostname matches the hosted zones accessible with the credentials and config). The strategy used will determine the contents of the DNSRecord resources Endpoints array.
+
+#### Simple
+
+```yaml
+apiVersion: kuadrant.io/v1alpha1
+kind: DNSRecord
+spec:
+  providerRef:
+    name: my-route53-credentials
+    namespace: my-gateways
+    kind: Secret
+  endpoints:
+    - dnsName: myapp.mn.hcpapps.net
+      recordTTL: 60
+      recordType: A
+      targets:
+        - 172.31.200.0
```

Comment:
How will the `simple` strategy look for Multi-cluster deployment? `A` record for each cluster

Reply:
No in an hub it would create a multi value A record

Reply:
@pehala Added example https://github.com/Kuadrant/architecture/pull/30/commits/3fef6f3e4cb9ac8587b18f89e5a8f5c6e1104ea1